### PR TITLE
Use pooled `StringBuilder` to reduce allocations when adding response cookies

### PR DESCRIPTION
**Pull request by dougbu**
- #561

### CODE REVIEW

File: src/Microsoft.Net.Http.Headers/SetCookieHeaderValue.cs [hunk 1]
```diff
@@ -90,42 +90,53 @@ public string Value
...
+        }
+
+        /// <summary>
+        /// Add string representation of this <see cref="SetCookieHeaderValue"/> to given <paramref name="builder"/>.
+        /// </summary>
+        /// <param name="builder">
+        /// The <see cref="StringBuilder"/> to receive the string representation of this
+        /// <see cref="SetCookieHeaderValue"/>.
+        /// </param>
+        public void AddStringValue(StringBuilder builder)
```
Comment (Tratcher) [inline src/Microsoft.Net.Http.Headers/SetCookieHeaderValue.cs]: This name is confusing. `ToStringBuilder`? `CopyToStringBuilder`? `SerializeToStringBuilder`? `AppendToStringBuilder`?

Comment (dougbu) [inline src/Microsoft.Net.Http.Headers/SetCookieHeaderValue.cs]: Sure, let's go w/ `AppendToStringBuilder`.

**Comment by Tratcher** [comment]
https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.CookiePolicy/CookiePolicyMiddleware.cs#L29 will also need to be updated.

Looks OK. Check with @davidfowl on the new hosting service.

**Comment by dougbu** [comment]
> [`CookiePolicyMiddleware`'s `ResponseCookieFeature` instantiation](https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNetCore.CookiePolicy/CookiePolicyMiddleware.cs#L29) will also need to be updated

That line is fine because `ResponseCookieFeature` has the same constructor it did before. I've successfully built aspnet/KestrelHttpServer and aspnet/Security with this and aspnet/Hosting#652 in place.

**Comment by dougbu** [comment]
When using this branch and aspnet/Hosting#652, overall allocations go down approximately 1.7%. Big savings in `char[]` and `StringBuilder` instances:

###### Before
dev allocations

###### After
maker allocations

maker builders

**Comment by dougbu** [comment]
Will come back to the pooling once we've got a few other ducks in line.

**Pull request closed without merging**